### PR TITLE
PHPLIB-1031: Test mongocryptd is not spawned when shared library is loaded

### PR DESCRIPTION
**Pull request by alcaeus**
PHPLIB-1031

This is mostly copy-pasta from previous tests, changing the skip condition at the top to only skip the test when crypt_shared isn't available.

### CODE REVIEW

File: tests/SpecTests/ClientSideEncryptionSpecTest.php [hunk 1]
```diff
@@ -1029,6 +1029,43 @@ static function (self $test, ClientEncryption $clientEncryption, ClientEncryptio
...
+        ];
+
+        $clientEncrypted = static::createTestClient(null, [], ['autoEncryption' => $autoEncryptionOpts]);
+
+        $clientEncrypted->selectCollection('db', 'coll')->insertOne(['unencrypted' => 'test']);
+
+        $clientMongocryptd = static::createTestClient('mongodb://localhost:27021');
+
+        $this->expectException(ConnectionTimeoutException::class);
+        $clientMongocryptd->selectDatabase('db')->command(['ping' => 1]);
```
Comment (jmikola) [inline tests/SpecTests/ClientSideEncryptionSpecTest.php:1066]: Calling `$clientMongocryptd->getManager()->selectServer()` would be the most direct way to trigger a handshake. If you stick with a command, I think `hello` might be more correct (prose tests says "run a handshake command"). I have no idea if mongocryptd supports `ping`, although that's arguably a moot point as the test will never get that far.

```suggestion
        $clientMongocryptd->selectDatabase('db')->command(['hello' => 1]);
```

File: tests/SpecTests/ClientSideEncryptionSpecTest.php [hunk 2]
```diff
@@ -1029,6 +1029,43 @@ static function (self $test, ClientEncryption $clientEncryption, ClientEncryptio
...
+            ],
+        ];
+
+        $clientEncrypted = static::createTestClient(null, [], ['autoEncryption' => $autoEncryptionOpts]);
+
+        $clientEncrypted->selectCollection('db', 'coll')->insertOne(['unencrypted' => 'test']);
+
+        $clientMongocryptd = static::createTestClient('mongodb://localhost:27021');
+
+        $this->expectException(ConnectionTimeoutException::class);
```
[the remark this answers is not in the document]
Comment (alcaeus) [reply tests/SpecTests/ClientSideEncryptionSpecTest.php:1065]: This was shamelessly copied from `testBypassSpawningMongocryptdViaBypassAutoEncryption` and is also used in `testBypassSpawningMongocryptdViaBypassQueryAnalysis`. I'll update all of these tests accordingly.

File: tests/SpecTests/ClientSideEncryptionSpecTest.php (hunk 1 above)
Comment (alcaeus) [reply tests/SpecTests/ClientSideEncryptionSpecTest.php:1066]: Using `selectServer` sounds good, as that will definitely trigger a handshake without relying on a specific command.

File: tests/SpecTests/ClientSideEncryptionSpecTest.php (hunk 2 above)
[the remark this answers is not in the document]
Comment (jmikola) [reply tests/SpecTests/ClientSideEncryptionSpecTest.php:1065]: The best criticism is self-criticism: https://github.com/mongodb/mongo-php-library/commit/af419eb8869a0903f5388f20d1891f287c491903

> I'll update all of these tests accordingly.

SGTM! See if `getManager()->selectServer()` makes more sense there as well.

[the remark this answers is not in the document]
Comment (alcaeus) [reply tests/SpecTests/ClientSideEncryptionSpecTest.php:1065]: Updated to use `selectServer`.

**Review by jmikola — APPROVED** [review]

File: tests/SpecTests/ClientSideEncryptionSpecTest.php [hunk 3]
```diff
@@ -1029,6 +1029,43 @@ static function (self $test, ClientEncryption $clientEncryption, ClientEncryptio
...
+            'keyVaultNamespace' => 'keyvault.datakeys',
+            'kmsProviders' => [
+                'local' => ['key' => new Binary(base64_decode(self::LOCAL_MASTERKEY), 0)],
+            ],
+            'schemaMap' => [
+                'db.coll' => $this->decodeJson(file_get_contents(__DIR__ . '/client-side-encryption/external/external-schema.json')),
+            ],
+            'extraOptions' => [
+                'mongocryptdBypassSpawn' => true,
+                'mongocryptdURI' => 'mongodb://localhost:27021/db?serverSelectionTimeoutMS=1000',
```
[the remark this answers is not in the document]
Comment (kevinAlbs) [reply tests/SpecTests/ClientSideEncryptionSpecTest.php:1053]: No. It can be removed.

[the remark this answers is not in the document]
Comment (jmikola) [reply tests/SpecTests/ClientSideEncryptionSpecTest.php:1053]: Created https://github.com/mongodb/specifications/pull/1417.

**Review by jmikola — APPROVED** [review]

**Pull request merged**